test/common/upstream/hds_test.cc

```diff
@@ -313,6 +374,95 @@ TEST_F(HdsTest, TestProcessMessageMissingFieldsWithFallback) {
   EXPECT_EQ(hds_delegate_friend_.getStats(*hds_delegate_).requests_.value(), 2);
 }
 
+// Test if sendResponse() retains the structure of all endpoints ingested in the specifier
+// from onReceiveMessage(). This verifies that all endpoints are grouped by the correct
+// cluster and the correct locality.
+TEST_F(HdsTest, TestSendResponseMultipleEndpoints) {
+  // number of clusters, localities by cluster, and endpoints by locality
+  // to build and verify off of.
+  const int NumClusters = 2;
```

Comment:
Nit: prefer unsigned explicit sized types, e.g. `uint32_t`.

Reply:
Thanks for the note. I went ahead and replaced the integer variables I added with an explicitly-sized type.